jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1849,7 +1850,8 @@ void getCharsNoBoundChecks(int start, int end, char[] data, int index) {
 	}
 
 	void getChars(int start, int end, byte[] data, int index) {
-		if (0 <= start && start <= end && end <= lengthInternal()) {
+		// {@code data.length} comes with a null check
+		if (0 <= start && start <= end && end <= lengthInternal() && 0 <= index && end - start <= data.length/2 - index) {
```

Comment:
this check doesn't seem right - for a decompressed string vs a compressed string you have a different min length requirement. Given this is package protected do we need this check or is a simple getClass or whatever sufficient to trigger the appropriate NPE?

Reply:
@andrewcraik Since this is package protected, maybe we don't need any checks? The caller should guarantee that the arguments passed in are valid.

Reply:
It can be called from the general JCL (eg other stuff in java/lang) so not sure - maybe @DanHeidinga has a better idea

Reply:
I think we should remove this `byte[]` overloaded API. It was formerly used in StringBuilder and StringBuffer but we have migrated all uses to `getCharsNoBoundChecks`, so there are to the best of my knowledge no known uses of this overload of `getChars` in the JCL. I vote we deprecate it here.